packages/manager/apps/pci-databases-analytics/src/pages/services/create/_components/ApiTerraformDialog.component.tsx

```diff
@@ -0,0 +1,335 @@
+import { useLocale } from '@/hooks/useLocale';
+
+import * as TTerraform from '@/types/terraform';
+import * as database from '@/types/cloud/project/database';
+import { Disk } from '@/types/cloud/project/database/service/Disk';
+import { LINKS, getDocumentationUrl } from '@/configuration/documentation';
+
+type OrderFunnelModel = ReturnType<typeof useOrderFunnel>;
+
+const ApiTerraformDialog = ({
+  opened,
+  onRequestClose,
+  dialogData,
+}: {
+  opened: boolean;
+  onRequestClose: () => void;
+  dialogData: OrderFunnelModel;
+}) => {
+  const { t } = useTranslation('pci-databases-analytics/services/new');
+  const projectData = usePciProject();
+  const locale = useLocale();
+  const { toast } = useToast();
+
+  const [apiBodyContent, setApiBodyContent] = useState<{ [key: string]: any }>(
+    {},
+  );
+  const [apiEndpoint, setApiEndpoint] = useState<string>('');
+
+  // API and Terraform schemas description
+  const [apiSchemaDescription] = useState<{ [key: string]: any }>({
+    description: t('pciDatabasesAddCommandParametersInfoApiDescription'),
+    'nodesPattern.flavor': t(
+      'pciDatabasesAddCommandParametersInfoApiNetworkId',
+    ),
+    'nodesPattern.number': t('pciDatabasesAddCommandParametersInfoApiSubnetId'),
+    'nodesPattern.region': t('pciDatabasesAddCommandParametersInfoApiDiskSize'),
+    plan: t('pciDatabasesAddCommandParametersInfoApiNodesPatternFlavor'),
+    version: t('pciDatabasesAddCommandParametersInfoApiNodesPatternNumber'),
+    'disk.size': t('pciDatabasesAddCommandParametersInfoApiNodesPatternRegion'),
+    networkId: t('pciDatabasesAddCommandParametersInfoApiPlan'),
+    subnetId: t('pciDatabasesAddCommandParametersInfoApiVersion'),
+  });
+  const [terraformSchemaDescription] = useState<{ [key: string]: any }>({
+    service_name: t('pciDatabasesAddCommandParametersInfoTerraformServiceName'),
+    description: t('pciDatabasesAddCommandParametersInfoTerraformDescription'),
+    engine: t('pciDatabasesAddCommandParametersInfoTerraformEngine'),
+    version: t('pciDatabasesAddCommandParametersInfoTerraformVersion'),
+    plan: t('pciDatabasesAddCommandParametersInfoTerraformPlan'),
+    flavor: t('pciDatabasesAddCommandParametersInfoTerraformFlavor'),
+    disk_size: t('pciDatabasesAddCommandParametersInfoTerraformDiskSize'),
+    nodes: t('pciDatabasesAddCommandParametersInfoTerraformNodes'),
+    'nodes.region': t(
+      'pciDatabasesAddCommandParametersInfoTerraformNodesRegion',
+    ),
+    'nodes.network_id': t(
+      'pciDatabasesAddCommandParametersInfoTerraformNodesNetworkId',
+    ),
+    'nodes.subnet_id': t(
+      'pciDatabasesAddCommandParametersInfoTerraformNodesSubnetId',
+    ),
+  });
+
+  const [terraformData, setTerraformData] = useState<
+    TTerraform.cloud.Response['resource'] | null
+  >('');
+  const { serviceToTerraform } = useServiceToTerraform({
+    onError: () => {},
+    onSuccess: (data) => {
+      setTerraformData(data?.resource);
+    },
+  });
```

Comment:
Use the whole response as an object, it might be useful in the future.
Also, you're missing the handler for onError, maybe add a toast to display api error
```suggestion
  const [
    terraformData,
    setTerraformData,
  ] = useState<TTerraform.cloud.Response | null>(null);
  const { serviceToTerraform } = useServiceToTerraform({
    onError: () => {},
    onSuccess: (data) => {
      setTerraformData(data);
    },
  });
```